dessia_common/exports.py

```diff
@@ -253,3 +256,100 @@ def autosize_sheet_columns(sheet, min_width=5, max_width=30):
+
+        all_class_names = list(set(subvalue.__class__.__name__ for subvalue in in_values))
+        if len(all_class_names) == 1:
+            printed_string += ''.join([f"{all_class_names[0]}",
+                                       f"{'s' if len(value) > 1 else ''}"])
+        else:
+            str_all_class = str(all_class_names).translate(str(all_class_names).maketrans('', '', "{}'"))
+            printed_string += f"{str_all_class}"
+
+        return printed_string
+
+    def _object_to_str(self, value) -> str:
+        if hasattr(value, 'name' ):
+            return f"{value.name}"
+        return 'unnamed'
+
+    def _value_to_str(self, value: Any) -> str:
+        if isinstance(value, (float, int, bool, complex)):
+            return str(round(value, 6))
+        if isinstance(value, str):
+            return (value if value != '' else 'no_value')
+        if isinstance(value, (list, dict, set)):
+            return self._sequence_to_str(value)
+        return self._object_to_str(value)
+
+    def _string_in_table(self, string: str = ''):
+        return string[:self.print_limit] + ('...' if len(string) > self.print_limit else '')
+
+    def _object_matrix(self, object_):
+        matrix = []
+        for attr, value in object_.__dict__.items():
+            matrix.append([attr,
+                           type(value).__name__,
+                           # (self._sequence_to_str(value) if isinstance(value, (list, dict, set)) else ' - '),
+                           (self._value_to_str(value) if not isinstance(value, (list, dict, set)) else ' - ')])
+        return matrix
+
+    def _head_table(self, col_names: List[str] = None) -> str:
+        cap_names = map(lambda x: x.capitalize(), col_names)
+        return ("| " + " | ".join(cap_names) + " |\n" +
+                "| ------ " * len(col_names) + "|\n")
+
+    def _table_line(self, row: List[Any]) -> str:
+        line = "|"
+        for value in row:
+            line += ''.join([" ", self._string_in_table(self._value_to_str(value)), " |"])
+        return line + "\n"
+
+    def _content_table(self, content: List[List[Any]]) -> str:
+        half_table = int(self.table_limit / 2)
+        table = ''
+        for row in content[:half_table]:
+            table += self._table_line(row)
+
+        if len(content) > self.table_limit and self.table_limit != -1:
+            table += f"| + {len(content) - self.table_limit} unprinted elements | |\n"
+
+        if len(content) > half_table:
+            for row in content[-half_table:]:
+                table += self._table_line(row)
+
+        return table
+
+    def print_name(self, object_) -> str:
+        return (object_.name if object_.name != '' else 'with no name')
+
+    def print_class(self, object_) -> str:
+        return object_.__class__.__name__
+
+    def matrix_table(self, matrix: List[List[float]], col_names: List[str] = None) -> str:
+        return ''.join([self._head_table(col_names),
```

Comment:
Does this line really need to be cut in half ? ie. doesn't it fit in the 120 characters limit  if kept as a one-liner ?

Reply:
Yes it does but I find it more readable

Reply:
I guess we all have different sensibilities to it. In fact, one or two years ago, I would have totally agreed with you, but right now I find it totally barbaric. I get the fact that they are spaced into a column, though which make total sense